spec/lexical-structure.md

```
@@ -0,0 +1,408 @@
+- `snake_case` for variables, functions, and module names
+- `PascalCase` for type names (structs, enums, interfaces)
+- `SCREAMING_SNAKE_CASE` for constants
+- `_leading_underscore` for private or internal entities
+
+## Keywords
+
+Keywords are reserved words that have special meaning in the language and cannot be used as identifiers except when escaped with backticks.
+
+### Reserved Keywords
+
+```
```

markdownlint-cli2 (0.17.2)

Fenced code blocks should have a language specified
null

(MD040, fenced-code-language)